src/Http/Routing/src/Matching/DictionaryJumpTable.cs

```diff
@@ -29,14 +29,14 @@ internal class DictionaryJumpTable : JumpTable
             }
         }
 
-        public override int GetDestination(string path, PathSegment segment)
+        public override int GetDestination(ReadOnlySpan<char> path)
         {
-            if (segment.Length == 0)
+            if (path.Length == 0)
             {
                 return _exitDestination;
             }
 
-            var text = path.Substring(segment.Start, segment.Length);
+            var text = path.ToString();
```

Comment:
@GrabYourPitchforks @stephentoub @jkotas it would be great to allow lookup by `ReadOnlySpan<char>` without the added allocation (closest issue I saw was this https://github.com/dotnet/runtime/issues/27229)

Reply:
Yes. The sticking point has been with a design regarding custom comparers:
https://github.com/dotnet/runtime/issues/28368#issuecomment-631059968

Reply:
One other sticking point is that there's no guarantee that the same algorithm is used to compute the hash code of a `string` vs. the hash code of a `ReadOnlySpan<char>`. In fact, `Dictionary<string, ...>` _throws away_ the default comparer and uses a specialized string comparer for greater performance. It's definitely solvable, but doing this in a manner that won't regress performance of _existing_ scenarios is a much more complex problem than it appears.

Reply:
I get it but how do we make progress? Does somebody just have to ask again? Do we have proposals?

Reply:
My proposal is what's in my linked comment: throw from the method of the comparer being used is not one we implicitly understand and can handle internally appropriately. It's not perfect, it's not beautiful, but it addresses what I think is the 99% case: using one of the built-in compares. The downside is there's a big cliff that manifests as either an exception or string allocation when the compared isn't supported. If we went with exception, we could make it work in the future if we added another interface supporting spans we could query for.

Reply:
There are a few candidate proposals in https://github.com/dotnet/runtime/issues/27229. (But Steve, I don't see your idea reflected there? Maybe I'm looking at the wrong piece?) The issue is currently marked as needs work and has no assignee. Somebody would need to champion it and drive it through API review.